orc8r/gateway/go/services/bootstrapper/service/bootstrapper.go

```diff
@@ -86,17 +86,37 @@ func NewLocalBootstrapper(bootstrapCompletionChan chan interface{}) *Bootstrappe
 
 // Initialize loads HW ID & challenge key and verifies it's validity
 func (b *Bootstrapper) Initialize() error {
```

Comment:
If you want an optional path, just do what snowflake already does - use variadic function:

Initialize(args ...string) ...

Reply:
I prefer the explicit arg enumeration, since the variadic + pseudo-untyped "args" is a little too much magic for my taste. But thanks for pointing out :)

Reply:
> Thanks for pointing out 🙂 I prefer the explicit arg enumeration, since the variadic + pseudo-untyped "args" is a little too much magic for my taste

It's not untyped, it's a string or nothing. Initialize("") could be more confusing & error prone.

Both ways are acceptable, IMO. I just think that the variadic function is a bit more intuitive for an optional argument.
Most use cases (all of them until now) are using default snowflake path & "Initialize()" seems like the most straightforward way to use it vs. "Initialize("")" which needs an additional explanation of the empty path string logic.
Only one use case so far (unit tests) needs custom path, so - I'd optimize API for the most common/simple case.

Go snowflake is a port of python's snowflake and the python version uses default argument for the path
https://github.com/shaddi/snowflake/blob/master/snowflake.py#L7, hence the decision to make the API to at least look similar.